### DISCUSSION TITLE
version.NEXT or v2 release timing and focus

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#719] [General] [answered]
I think if I recall v2 was going to be released around the time of .NET 5 release?

I just want to get an idea of when certain "breaking" changes are going to be allowed into the community project.

I know v1 is focused primarily on functionality in these areas:

> V1.0.0 (MVP)
> 
>  Multi-Tenant ( Shared Database & Isolated Database )
>  Modular Architecture / Headless API
>  Dynamic Page Compositing Model / Site & Page Management
>  Authentication / User Management / Profile Management
>  Authorization / Roles Management / Granular Permissions
>  Dynamic Routing
>  Extensibility via Custom Modules
>  Extensibility via Custom Themes
>  Event Logging
>  Folder / File Management
>  Recycle Bin
>  Scheduled Jobs ( Background Processing )
>  Notifications / Email Delivery
>  Auto-Upgrade Framework

v2 takes this even further however building out security features and device friendly features may also be a focus point for developers to think about for possibilities.  

Anyhow, just trying to look to the future for any possible chances to resolve any current or upcoming issues and take advantage of that moment in time when it comes.

> V.Next ( still in the process of being prioritized )
> 
> Admin UI markup optimization
> DB Migrations for framework installation/upgrade
> Support for SQLite
> Static Localization ( ie. labels, help text, etc.. )
> Migrate to Code-Behind Pattern ( *.razor.cs )
> Generic Repository Pattern
> JwT token authentication ( possibly using IdentityServer )
> Optional Encryption for Settings Values

I guess I am looking into these to where maybe we can put together a discussion and people interested in helping contribute in those areas can add to that discussion focusing on talking points to help guide community developers on what to research and develop for possible community accepted solutions what do you all think?

**Comment by sbwalker** [comment, marked as answer]
I think people did not reply to this message because there is already a similar discussion happening in #709
